src/redisai.c

```diff
@@ -1332,6 +1346,12 @@ int RedisAI_Config_BackendsPath(RedisModuleCtx *ctx, const char *path) {
   return RedisModule_ReplyWithSimpleString(ctx, "OK");
 }
 
+int RedisAI_Config_QueueThreads(RedisModuleCtx *ctx, const char *queueThreadsString) {
+  RedisModule_AutoMemory(ctx);
+  perqueueThreadPoolSize = atoi(queueThreadsString);
```

Comment:
Also you can pass the RedisModuleString directly and use RedisModule_StringToLongLong

Reply:
Also the toNumber conversion might failed .. RedisModule_StringToLongLong will return if it was successful or not

Reply:
@MeirShpilraien and @gkorland followed your advice and changed it towards using RedisModule_StringToLongLong. Regarding the conversion to a number we also need to double-check it is a positive integer ( added it as well ) - otherwise, at the moment we allocate the threads for each device it would enter in a breaking state.